bin/sharepoint.py

```diff
@@ -34,6 +36,17 @@ def copy_file(local_path, target_path):
     with open(local_path, 'rb') as content_file:
         file_content = content_file.read()
     
+    #test for empty df if the data is a CSV (may be other types like shapefile or txt)
+    if file_name.split('.')[-1] == 'csv':
+        df = pd.read_csv(BytesIO(file_content))
```

Comment:
to avoid reading in the whole file, you can add nrows=10:
 df = pd.read_csv(BytesIO(file_content),nrows=10)
it won't give an error if they file is empty